### ISSUE TITLE
383: rpm -e and verify conflicting files removed 1   FAILED (rpme.at:327)

### ISSUE THREAD

**Issue by JetXujing** [rpm-software-management/rpm#3242]
I ran test cases using rpm-4.18.2 and found one test case failed:
```
...
383: rpm -e and verify conflicting files removed 1   FAILED (rpme.at:327)
...
```

I encountered this problem in 4.18.0, but it was too late to provide a reproduction method, resulting in the issue(https://github.com/rpm-software-management/rpm/issues/2376) being closed.

The steps to reproduce it are as follows:
```
[root@localhost SPECS]# cat conflicttest.spec
# avoid depending on rpm configuration
%define _datadir /usr/share

Name:           conflict%{pkg}
Version:        1.0
Release:        1
Summary:        Testing file conflict behavior

Group:          Testing
License:        GPL
BuildArch:      noarch

%description
%{summary}

%install
mkdir -p $RPM_BUILD_ROOT/%{_datadir}
echo "%{filedata}" > $RPM_BUILD_ROOT/%{_datadir}/my.version

%files
%defattr(-,root,root,-)
%{?fileattr} %{_datadir}/my.version
```
1. rpmbuild -bb --quiet --define "pkg a" --define "filedata pkg-a" conflicttest.spec
2. rpmbuild -bb --quiet --define "pkg b" --define "filedata pkg-b" conflicttest.spec
3. rpm -U --ignoreos /root/rpmbuild/RPMS/noarch/conflicta-1.0-1.noarch.rpm
4. rpm -U --ignoreos --force /root/rpmbuild/RPMS/noarch/conflictb-1.0-1.noarch.rpm
5. rpm -Vav --nogroup --nouser conflicta conflictb

I get the following results that do not meet expectations:
```
[root@localhost noarch]# rpm -Vav --nogroup --nouser conflicta conflictb
.........    /usr/share/my.version
.........    /usr/share/my.version (not installed)
```
According to the test cases, the expected results are as follows:
```
[root@localhost noarch]# rpm -Vav --nogroup --nouser conflicta conflictb
.........    /usr/share/my.version (not installed)
.........    /usr/share/my.version
```

**Comment by dmnks** [comment]
OK, after looking into this more w/ @pmatilai, this turns out to be a test bug (obligatory [meme](https://programmerhumor.io/wp-content/uploads/2023/07/programmerhumor-io-testing-memes-debugging-memes-c242cb2722b41c1.jpg), thanks @pmatilai for sharing).

The culprit is that the tests use the `-a` option which causes the output lines to depend on the order in the rpmdb, not on the order of arguments given on the command line. The correct way to use `-V` here is *without* `-a`, then the order will reflect the arguments, as one (and certainly the tests) would expect.

The fix is easy, I'll do it shortly.

Just a note: The example output above corresponds to a different test in the same `rpme.at` file, not to the commands above it (but the point stands).

**Comment by dmnks** [comment]
Oh, I somehow didn't mention the most important part: The reason you've hitting this is most likely because you've configured rpm to use the `ndb` backend (instead of the default `sqlite`). With the former set, this is easily reproducible.

**Comment by JetXujing** [comment]
> Oh, I somehow didn't mention the most important part: The reason you're hitting this is most likely because you've configured rpm to use the `ndb` backend (instead of the default `sqlite`). With the former set, this is easily reproducible.

Thank you for your reply. Yes, I am using the ndb database.

**Comment by dmnks** [comment]
Fixed upstream via #3254. Please note, however, that we're not planning on backporting this to the 4.18.x branch.
